src/tamp_improv/approaches/improvisational/graph_training.py

```diff
@@ -267,10 +271,22 @@ def collect_graph_based_training_data(
 
         # Find potential shortcuts using the collected states
         print(f"\nIdentifying shortcuts using {len(observed_states)} observed states")
-        shortcut_candidates = identify_shortcut_candidates(
-            planning_graph,
-            observed_states,
-        )
+
+        if use_random_rollouts:
+            shortcut_candidates = identify_promising_shortcuts_with_rollouts(
+                system,
+                planning_graph,
+                observed_states,
+                num_rollouts_per_node,
+                max_steps_per_rollout,
+                shortcut_success_threshold,
+                random_seed=seed + episode,  # Use different seed for each episode
```

Comment:
good catch. let's go a step further...

```
from tomsutils.utils import sample_seed_from_rng
rng = np.random.default(seed)
for episode in range(collect_episodes):
    episode_seed = sample_seed_from_rng(rng)
```
the reason that this is potentially better is that if we later run overall experiments with seed=0, seed=1, seed=2, we don't want to introduce unintentional correlation between the experiments by using the same random rollouts between seeds, which would happen if we used the current code